go/private/BUILD.sdk.bazel

```diff
@@ -47,7 +47,7 @@ filegroup(
 go_sdk(
     name = "go_sdk",
     srcs = [":srcs"],
-    experiments = {experiments},
+    experiments = "{experiments}",
```

Comment:
Let's keep using `repr` on the joined string so that we don't need to manually add quotes.

Reply:
This matches `goarch`, `goos`, and `version`. I'm fine with using `repr` but we should we swap them all for consistency? (And then it's a bit unclear what to do with `go = "bin/go{exe}",`

Reply:
`repr` is preferable for `experiments` as its value is directly set by users and may thus mess with manual escaping. We can leave the other attributes as is for now or also move them to `repr`. `{exe}` could be replaced with logic in the repo rules.

Reply:
gotcha, I noticed we were already using `repr` on `exec_compatible_with` so I only changed `experiments` for now; `goos` and `gooarch` are also used to build paths for a glob